### DISCUSSION TITLE
Relationships between animation-based game systems (animation, animscene, task, task sequence, scenario)

### DISCUSSION THREAD

**Discussion by Roms1383** [Halen84/ScriptHookRDR2DotNet-V2#11] [Q&A] [answered]
Hi @Halen84 ! Following your answer on #10 I'm gonna avoid solliciting you too much, but I think RDR really deserves some explanations and your repo seems like the perfect place to gather informations for newcomers.

Looking at the SDK, we have `animation` / `animscene` / `task` (and `task sequence`) / `scenario` : could you briefly describe how they differ or relate ?

I've only tried a couple `TASK` so far and they seems like short-lived tasks and I guess they can be combined with `SEQUENCE_TASK`. The fact that there's a [TASK_SCRIPTED_ANIMATION](https://www.rdr2mods.com/nativedb/index/task/task_scripted_animation/) also makes me wonder how task and animation relates.

Looking at `SCENARIO` I could find a list [there](https://www.rdr2mods.com/wiki/pages/list-of-rdr2-scenarios) but I can't really get how they differ from a task, if only involving props maybe ? (since for example there's `WORLD_PLAYER_CHORES_SACK_PICKUP_1` and `WORLD_PLAYER_CHORES_SACK_PUT_DOWN_1` which indicates a ped can pickup a sack and put it down elsewhere, which to me looks like a task of some sort).

I think the disturbing part being also being how `animation` and `animscene` relates to each other.

_Note: many of these methods I have yet to try by myself so I'll also report what I found out here._

**Comment by Halen84** [comment, marked as answer]
Animations and scenarios are both technically tasks.

## Scenario's
A scenario is just a point (scenario point) on the map where ped's can go and do certain animations, e.g. smoking, sweeping, reading newspaper, etc. Scenario points and what ped's can play them are defined in the game files, so the scenarios are not random.

## Task Sequence's
A task sequence is just a way for ped's to perform multiple tasks in order. It's usage goes like this (C++):
```cpp
int sequenceId = 0;
TASK::OPEN_SEQUENCE_TASK(&sequenceId);

// Now we give this sequence some tasks to execute.
// NOTE (IMPORTANT): The ped that you have opened a task sequence for must be passed in as 0 or NULL for the task parameters or else the task wont be added to the sequence. 
TASK::TASK_TURN_PED_TO_FACE_ENTITY(0, entity, 1000, -1.0f, -1.0f, -1.0f);
TASK::TASK_PLAY_ANIM(0, animDict, animName, 8.0f, -8.0f, -1, 0, 0.0f, false, 0, false, 0, false);

// Close the sequence task. No more tasks can be added to the sequence at this point.
TASK::CLOSE_SEQUENCE_TASK(sequenceId);

// Perform sequence task.
TASK::TASK_PERFORM_SEQUENCE(myPed, sequenceId);

// Flushes all tasks in the sequence. This should be called when the sequence task is done and no longer needed.
TASK::CLEAR_SEQUENCE_TASK(&sequenceId);
```
## AnimScene's
A AnimScene (animation scene) is basically just a set of multiple animations that will be played on the ped. In-game cutscenes and picking plants for example are AnimScenes. These can also be found in the game files. AnimScene file extension is `.yas` and can be opened with programs like OpenIV or CodeX

**Reply by Halen84** [reply]
Also, use [alloc8or's natives database](https://alloc8or.re/rdr3/nativedb/) when researching natives. It contains the most up-to-date list.

Also be sure to check out [femga's repo](https://github.com/femga/rdr3_discoveries). Contains lots of random but useful stuff.

**Reply by Halen84** [reply]
If you want, you can join the rdr2mods.com [discord server](https://discord.com/invite/KSU7vFx) for real time help in the #dev-talk channel
